packages/react-testing/src/root.tsx

```diff
@@ -107,6 +107,8 @@ export class Root<Props> {
     });
 
     if (isPromise(result)) {
+      updateWrapper();
```

Comment:
Hmm, I'm not sure I understand why you would do it here, also. if the result is a promise, there shouldn't be a reason to update until after the act block has finished, is there?

Reply:
If the action you take looks something like

```
async function doASubmit() { 
   setSomeLoadingState(true);
   await sendSomeRequest();
   setSomeLoadingState(false);
}
```

Then after synchronous execution pauses on the await there has been a state update, if we don't call `updateWrapper` then any test for loading states in cases will fail.

Reply:
For this to make sense, `setSomeLoadingState` has to run synchronously. I don't think that's guaranteed for async functions?

Reply:
Resolved in Slack, we agreed that this is probably the expected behaviour, and if they need to wait until any other update is finished and then update the tree, they can do so with `act`